### ISSUE TITLE
on item hit event does not save the item correctly

### ISSUE THREAD

**Issue by Govindass** [SkQuery/SkQuery#54]
**Describe the bug**
in the on item hit event, the data value is not persisted on 1.12.2, so if it's spawn egg for example, it'll use 383:0 instead of 383:any data it has

**To Reproduce**
```
make player shoot an item projectile of zombie spawn egg at speed 1
on item hit:
  broadcast "%event-item%" #will return bat spawn egg
```
**Information:**
- Skript version: 2.5
- Server software: Paper
- Server version: 1.12.2

**Additional context**
None

**Comment by TheLimeGlass** [comment]
Can't reproduce on 1.16 with Skript 2.5

**Comment by Govindass** [comment]
> 
> 
> Can't reproduce on 1.16 with Skript 2.5

might be 1.12.2 only bug

**Comment by Govindass** [comment]
it's now fixed in 4.1.4, thank you!
